benchmarks/forest.jl

```diff
@@ -0,0 +1,130 @@
+using Agents
+using BenchmarkTools
+using Random
+
+# Define new space model
+mutable struct Tree <: AbstractAgent
+    id::Int
+    pos::Tuple{Int,Int}
+    status::Bool  # true is green and false is burning
+end
+function forest_fire_array(; f = 0.02, d = 0.8, p = 0.01, griddims = (100, 100), seed = 111)
+    Random.seed!(seed)
+    space = ArraySpace(griddims, true)
+    properties = Dict(:f => f, :d => d, :p => p)
+    forest = AgentBasedModel(Tree, space; properties = properties)
+
+    ## create and add trees to each pos with probability d,
+    ## which determines the density of the forest
+    for pos in positions(forest)
+        if rand() ≤ forest.d
+            add_agent!(pos, forest, true)
+        end
+    end
+    return forest, dummystep, forest_model_step_array!
+end
+
+function forest_model_step_array!(forest)
+    for pos in positions(forest, :random)
+        nc = get_node_contents(pos, forest)
+        ## the cell is empty, maybe a tree grows here
+        if length(nc) == 0
+            rand() ≤ forest.p && add_agent!(pos, forest, true)
+        else
+            tree = forest[nc[1]] # by definition only 1 agent per pos
+            if tree.status == false  # if it is has been burning, remove it.
+                kill_agent!(tree, forest)
```

Comment:
Is `kill_agent!` updated to use the new `remove_agent_from_space!` function?

Reply:
Yes: https://github.com/JuliaDynamics/Agents.jl/blob/master/src/core/space_interaction_API.jl#L63-L67

`kill_agent!` is now space agnostic, same as `genocide!`.

Reply:
I just found that it is.